git_theta/api.py

```diff
@@ -0,0 +1,71 @@
+"""User facing functions to interact with git-theta without incurring extra io costs."""
+
+import io
+import sys
+from typing import Optional
+
+import git
+
+import git_theta
+from git_theta import checkpoints, filters, git_utils, metadata, utils
+
+
+# TODO: Should we have a `materialize` flag that writes the model to disk so you
+# don't have git status showing a diff?
+def save(
+    state_dict,
+    path: str,
+    commit_msg: Optional[str] = None,
```

Comment:
Do we really want to make this optional? Git treats it as non-optional.

Reply:
We have a default value of `Committing ${path}`, I assume if we make it mandatory there would be a lot of `ups`/`update` commits.

I think the advantage of not having a default is that function sig basically matches torch.save, but we can remove the default if we think it will make people add more informative commits.

Reply:
Yeah, I think I am philosophically opposed to making the default to have an uninformative commit message - we want git-theta to be more than torch.save on steroids.

Reply:
I made it required